### PR TITLE
[Iwara] Improve metadata extraction

### PR DESCRIPTION
**Pull request by AlexAplin**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Adds upload_date, uploader, description, comment_count, like_count, and view_count metadata extraction.

### CODE REVIEW

**Comment by Rektis** [comment]
On the master branch, metadata extraction such as using %(uploader)s still doesn't work on iwara.tv.

**Comment by timofonic** [comment]
@AlexAplin  Any news?

**Comment by AlexAplin** [comment]
@dstftw @yan12125 Any chance of getting this merged or should I move on?

File: youtube_dl/extractor/iwara.py [hunk 1]
```diff
@@ -71,6 +81,26 @@ def _real_extract(self, url):
         title = remove_end(self._html_search_regex(
             r'<title>([^<]+)</title>', webpage, 'title'), ' | Iwara')
 
+        upload_date = unified_strdate(self._html_search_regex(
+            r'(\d{4}-\d{2}-\d{2})', webpage, 'upload_date', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Too ambiguous. This will just capture first date found.

File: youtube_dl/extractor/iwara.py [hunk 2]
```diff
@@ -71,6 +81,26 @@ def _real_extract(self, url):
...
+
+        uploader = get_element_by_class('username', webpage)
+
+        description_class = get_element_by_class('field-type-text-with-summary', webpage)
+        description = clean_html(description_class.replace('</p>', '<br /></p>') if description_class else None)
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: What's the point of this replace?

File: youtube_dl/extractor/iwara.py [hunk 3]
```diff
@@ -71,6 +81,26 @@ def _real_extract(self, url):
...
+
+        comment_count_classes = get_elements_by_class('title', webpage)
+        comment_count = None
+        if comment_count_classes and len(comment_count_classes) >= 2:
+            comment_count = int_or_none(''.join(digit for digit in comment_count_classes[1] if digit.isdigit()))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: `_search_regex` instead of this mess.

File: youtube_dl/extractor/iwara.py [hunk 4]
```diff
@@ -71,6 +81,26 @@ def _real_extract(self, url):
...
+
+        node_views_class = clean_html(get_element_by_class('node-views', webpage))
+        node_views = node_views_class.split() if node_views_class else None
+        like_count = view_count = None
+        if node_views and len(node_views) >= 2:
+            like_count = int_or_none(node_views[0].replace(',', ''))
+            view_count = int_or_none(node_views[1].replace(',', ''))
```
Comment (dstftw) [inline youtube_dl/extractor/iwara.py]: Again `_search_regex`.

**Pull request closed without merging**